aspnetcore/mvc/controllers/filters.md

```diff
@@ -277,147 +347,157 @@ The [short circuiting resource filter](#short-circuiting-resource-filter) shown
-* `Result` - setting this short-circuits execution of the action method and subsequent action filters. Throwing an exception also prevents execution of the action method and subsequent filters, but is treated as a failure instead of a successful result.
+* <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext.ActionArguments> - enables the inputs to an action method be read or modified.
+* <xref:Microsoft.AspNetCore.Mvc.Controller> - enables manipulating the controller instance.
+* <xref:System.Web.Mvc.ActionExecutingContext.Result> - setting `Result` short-circuits execution of the action method and subsequent action filters.
+
+Throwing an exception in an action method:
+
+* Prevents execution of the action method and subsequent filters.
+* Unlike setting `Result`, is treated as a failure instead of a successful result.
 
 The <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutedContext> provides `Controller` and `Result` plus the following properties:
 
-* `Canceled` - will be true if the action execution was short-circuited by another filter.
-* `Exception` - will be non-null if the action or a subsequent action filter threw an exception. Setting this property to null effectively 'handles' an exception, and `Result` will be executed as if it were returned from the action method normally.
+* <xref:System.Web.Mvc.ActionExecutedContext.Canceled> - True if the action execution was short-circuited by another filter.
+* <xref:System.Web.Mvc.ActionExecutedContext.Exception> - Non-null if the action or a subsequent action filter threw an exception. Setting this property to null:
+
+  * Effectively handles the exception.
+  * `Result` is executed as if it was returned from the action method.
 
-For an `IAsyncActionFilter`, a call to the `ActionExecutionDelegate`:
+For an `IAsyncActionFilter`, a call to the <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutionDelegate>:
 
 * Executes any subsequent action filters and the action method.
-* returns `ActionExecutedContext`. 
+* Returns `ActionExecutedContext`.
+
+To short-circuit, assign <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext.Result?displayProperty=fullName> to a result instance and don't call the `ActionExecutionDelegate`.
 
-To short-circuit, assign `ActionExecutingContext.Result` to some result instance and don't call the `ActionExecutionDelegate`.
+The framework provides an abstract <xref:Microsoft.AspNetCore.Mvc.Filters.ActionFilterAttribute> that can be subclassed.
 
-The framework provides an abstract `ActionFilterAttribute` that you can subclass. 
+Action filters can be used:
 
-You can use an action filter to validate model state and return any errors if the state is invalid:
+* To validate model state.
+* Return  errors if the state is invalid:
 
-[!code-csharp[](./filters/sample/src/FiltersSample/Filters/ValidateModelAttribute.cs)]
+[!code-csharp[](./filters/sample/FiltersSample/Filters/ValidateModelAttribute.cs?name=snippet)]
```

Comment:
Can this code be shifted over slightly? It's contextually relevant to the bullet above but the description that follows makes me expect to see `OnActionExecuted` in the code example.

Reply:
I added `OnActionExecuted` - check out what I have.

Reply:
Yeah, that looks good. It might be worth highlighting the `OnActionExecuted` code as the snippet contains both methods. Or perhaps the snippet can be updated to show just that method here.

Reply:
Yeah, that looks good. It might be worth highlighting the `OnActionExecuted` code as the snippet contains both methods. Or perhaps the snippet can be updated to show just that method here.